planex/makesrpm.py

```diff
@@ -39,45 +43,24 @@ def parse_args_or_exit(argv):
         "--keeptmp", action="store_true",
         help="keep temporary files")
     argcomplete.autocomplete(parser)
-    return parser.parse_args(argv)
 
+    parsed_args = parser.parse_args(argv)
+    links = [arg for arg in argv if arg.endswith(".lnk")]
+    parsed_args.link = None
+    if links:
+        parsed_args.link = links[0]
 
-def setup_tmp_area():
-    """
-    Create temporary working area
-    """
-    tmp_dirpath = tempfile.mkdtemp()
-    tmp_specs = os.path.join(tmp_dirpath, 'SPECS')
-    tmp_sources = os.path.join(tmp_dirpath, 'SOURCES')
-
-    os.makedirs(tmp_specs)
-    os.makedirs(tmp_sources)
-
-    return (tmp_dirpath, tmp_specs, tmp_sources)
+    patchqueues = [arg for arg in argv if arg.endswith("patches.tar")]
+    parsed_args.patchqueue = None
+    if patchqueues:
+        parsed_args.patchqueue = patchqueues[0]
 
-
-def extract_topdir(tmp_specfile, source):
-    """
-    Set the topdir name taken from the source tarball
-    """
-    for line in fileinput.input(tmp_specfile, inplace=True):
-        if 'autosetup' in line:
-            tar = tarfile.open(source)
-            names = tar.getnames()
-            topname = os.path.commonprefix(names)
-            if topname in names:
-                top_element = tar.getmember(topname)
-                if top_element.isdir():
-                    print "%s -n %s" % (line.strip(), topname)
-            else:
-                print "%s -c" % line.strip()
-        else:
-            print line,
+    return parsed_args
 
```

Comment:
Given the amount of changes above, I think we should improve the helpstrings generated for the tool, spec argument and source argument. Or maybe add link and patches.tar as explicit arguments.